models/songs.model.js

```diff
@@ -32,7 +34,30 @@ Song.findById = function (id, result) {
 
 Song.findAll = function (result) {
   dbConn.query(
-    "Select id,name,artist,duration,image,created_date from songs",
+    "Select id,name,artist,url,duration,image,album,created_date from songs",
+    function (err, res) {
+      if (err) {
+        result(null, err);
+      } else {
+        result(null, res);
+      }
+    }
+  );
+};
+
+Song.findAlbums = function (result) {
+  dbConn.query("Select album from songs group by album", function (err, res) {
+    if (err) {
+      result(null, err);
```

Comment:
shouldn't the error be passed as the first argument to the `result` function?

Reply:
we will be adding error handlers & global error handlers for all queries/endpoints in the next push